### ISSUE TITLE
Develop a system to flag candidates (Grey, Red, Green) based on their test progress and results

### ISSUE THREAD

**Issue by EhsanEhrari** [talentbeyondboundaries/tbbtalentv2#1580]
This system will flag candidates based on their test progress and results, categorizing them into one of three statuses: Gray, Red, Orange, or Green. The flagging will be visualized using a color-coded bubble and display additional information about the candidate's test results. 

**Flagging Criteria:**

- **Gray: Pending**
This status will be assigned when the candidate has not yet completed the test or the test results are not available. The flag will appear gray, indicating that the candidate's test status is still pending.

- **Red: Below Requirement (<60)**
If a candidate's test score is below 60, they will be flagged with a red bubble, indicating they are below the required language proficiency threshold. This indicates the candidate did not meet the minimum requirement for the test.

- **Orange: Needs Verification (60-90)**
If the candidate’s score falls between 60 and 90, it will be flagged orange, indicating that the score needs further verification against the required language proficiency. This is a warning state where additional review may be needed.

- **Green: Meets Requirement (90+)**
A candidate with a test score of 90 or higher will be flagged with a green bubble. This status means that the candidate meets or exceeds the required language proficiency, confirming their eligibility.

**Visualization & Features:**

- **Duo Owl Icon:**
Each candidate’s test result will be represented by a Duo owl icon that reflects the flag color (Gray, Red, Orange, or Green).
The icon will serve as a quick visual indicator of the candidate's language proficiency status.

- **Hover Over Scores:**
When the user hovers over the Duo owl icon, a tooltip will appear displaying the candidate's most recent score as well as their best score across all tests.
This provides more detailed insight into the candidate's performance without cluttering the interface.

- **Best and Most Recent Results:**
The best result will be the highest score a candidate has achieved in any language test.
The most recent result will be the latest test score the candidate has obtained.
Both the best and most recent results will be displayed to provide a clearer picture of the candidate’s progress and performance over time.

**Comment by EhsanEhrari** [comment]
On what overall score should the candidate be flagged as "Red"?

**Comment by EhsanEhrari** [comment]
What specific overall score threshold (e.g., >= 80) should be used to determine a "Green" flag?

**Comment by EhsanEhrari** [comment]
Do we want to store the candidate's flag persistently?
If yes, where should the flag be stored? In the Candidate entity, CandidateExam?
Or maybe, the better question is where do we want to display the Candidate Flag?

**Comment by sadatmalik** [comment]
From discussion with Zeynpe + team:

Gray: pending
<60: Red (below requirement)
60-90: Orange (needs verification against the language requirement)
90+: Green (meets language requirements)

Configurable as application.yaml properties

**Comment by sadatmalik** [comment]
Agreed the below in project meeting, if not possible to implement in this ticket, the dev will be handled in separate tickets. @EhsanEhrari to review and decide which approach is best:

Display bubble flag on the intake languages plan
Show a "Duo" owl icon with the colours + hover over scores
Show the Best and Most Recent results

**Comment by EhsanEhrari** [comment]
High-level diagram:

https://app.diagrams.net/#G1gjVeOVAaKcTBhk2k3tqPkNjMNkw8rpuU#%7B%22pageId%22%3A%22prtHgNgQTEPvFCAcTncT%22%7D

Updated diagram v2 (21-Nov-2024):
https://drive.google.com/file/d/1l3cSOd3XNjZVMvRF7oqyV4PQBELjhkdx/view?usp=sharing

**Comment by EhsanEhrari** [comment]
<img width="1031" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBek9DQ0E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--9faa91c100693716de9fd2c0e26f0e42c8f7ea71/image.png" alt="image.png" />
